hwbench/environment/vendors/mock.py

```diff
@@ -17,29 +18,56 @@ def read_thermals(
         self, thermals: dict[str, dict[str, Temperature]] = {}
     ) -> dict[str, dict[str, Temperature]]:
         # Let's add a faked thermal metric
-        thermals[str(ThermalContext.CPU)] = {"CPU1": Temperature("CPU1", 40)}
+        name = "CPU1"
+        if str(ThermalContext.CPU) not in thermals:
+            thermals[str(ThermalContext.CPU)] = {}
+        if name not in thermals[str(ThermalContext.CPU)]:
+            thermals[str(ThermalContext.CPU)][name] = Temperature(name)
+
+        thermals[str(ThermalContext.CPU)][name].add(40)
         return thermals
 
     def read_fans(
         self, fans: dict[str, dict[str, MonitorMetric]] = {}
     ) -> dict[str, dict[str, MonitorMetric]]:
         # Let's add a faked fans metric
-        fans[str(FanContext.FAN)] = {"Fan1": MonitorMetric("Fan1", "RPM", 40)}
+        name = "Fan1"
+        if str(FanContext.FAN) not in fans:
+            fans[str(FanContext.FAN)] = {}
+        if name not in fans[str(FanContext.FAN)]:
+            fans[str(FanContext.FAN)][name] = MonitorMetric(name, "RPM")
+
+        fans[str(FanContext.FAN)][name].add(40)
         return fans
 
     def read_power_consumption(
         self, power_consumption: dict[str, dict[str, Power]] = {}
     ) -> dict[str, dict[str, Power]]:
         # Let's add a faked power metric
-        power_consumption[str(PowerContext.BMC)] = {"Chassis": Power("Chassis", 125.0)}
+        name = str(PowerCategories.CHASSIS)
+        if str(PowerContext.BMC) not in power_consumption:
+            power_consumption[str(PowerContext.BMC)] = {}
+        if name not in power_consumption[str(PowerContext.BMC)]:
+            power_consumption[str(PowerContext.BMC)][name] = Power(
+                str(PowerCategories.CHASSIS)
+            )
```

Comment:
Optional: Maybe this `if context_name not in metrics_dict` / `if name not in metrics_dict[context_name]` logic can also be done in common here. Maybe even using a defaultdict.

Reply:
I read the commits in the wrong order, ignore this comment